packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -3485,6 +3487,13 @@ export class SnapController extends BaseController<
       }
     }
 
+    if (
+      origin !== METAMASK_ORIGIN &&
+      CLIENT_ONLY_HANDLERS.includes(handlerType)
+    ) {
+      throw new Error(`"${handlerType}" can only be invoked by MetaMask.`);
```

Comment:
Wondering if we should throw a proper JSON-RPC error here, instead of a generic error. I know we don't do it for other errors in this function either, but I think it makes sense to do?

Probably something for another PR to replace all errors though.

Reply:
Hmm, is it really expectation for a controller to throw JSON-RPC errors?

Reply:
But yeah definitely not feasible for this PR